game-app/game-core/src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -186,49 +184,83 @@ private String newMapUrlAndSizeLabelText(final List<MapDownloadItem> selectedMap
     if (selectedMapItems.isEmpty()) {
       return "<html>None selected</html>";
     }
-    final @NonNls String doubleSpace = "&nbsp;&nbsp;";
 
     String mapsString =
         String.join(", ", selectedMapItems.stream().map(MapDownloadItem::getMapName).toList());
 
     final StringBuilder sb = new StringBuilder();
     sb.append("<html>")
         .append(MessageFormat.format("Selected: {0}", mapsString))
-        .append(doubleSpace);
+        .append("&nbsp;&nbsp;");
 
-    if (selectedMapItems.size() == 1) {
-      final MapDownloadItem map = selectedMapItems.get(0);
-      if (!downloadMapsWindowModel.isInstalled(map)) {
-        if (map.getDownloadSizeInBytes() != -1L) {
-          sb.append(doubleSpace)
-              .append(" (")
-              .append(FileUtils.byteCountToDisplaySize(map.getDownloadSizeInBytes()))
-              .append(")");
-        }
-      } else {
-        downloadMapsWindowModel
-            .getInstallLocation(map)
-            .ifPresent(
-                mapPath -> {
-                  sb.append(doubleSpace).append(" (");
-                  try {
-                    sb.append(FileUtils.byteCountToDisplaySize(Files.size(mapPath)));
-                  } catch (final IOException e) {
-                    log.warn("Failed to read file size", e);
-                    sb.append("N/A");
-                  }
-                  sb.append(")")
-                      .append("<br>")
-                      .append(MessageFormat.format("Path: {0}", mapPath.toAbsolutePath()));
-                });
-      }
-    }
+    appendSizeAndPath(selectedMapItems, sb);
     sb.append("<br>");
     sb.append("</html>");
 
     return sb.toString();
   }
 
+  private void appendSizeAndPath(
+      final List<MapDownloadItem> selectedMapItems, final StringBuilder sb) {
```

Comment:
Do you have any thoughts to rewrite these methods to have no side effects?

Reply:
I did not recognize any even when selecting multiple maps with >400MB

Reply:
`private void` methods are arguably evil. Namely because they are implicitly stateful.

These methods are calculating multiple things. To make the methods functional, mark them as static and have them return the things calculated. Since we are calculating multiple things - have multiple methods. Try to keep control-flow logic in the "top most method" as much as possible as well.

Reply:
Please check after new commit https://github.com/triplea-game/triplea/pull/13516/commits/5abf2a1359eb7eb29a280f85b22462fe0076d805

Reply:
Added comments to the commit